gradle-plugin/src/main/groovy/io/realm/gradle/Realm.groovy

```diff
@@ -44,27 +42,36 @@ class Realm implements Plugin<Project> {
         def syncEnabledDefault = false
         project.extensions.create('realm', RealmPluginExtension, project, syncEnabledDefault)
 
-        def usesKotlinPlugin = project.plugins.findPlugin('kotlin-android') != null
         def usesAptPlugin = project.plugins.findPlugin('com.neenbedankt.android-apt') != null
+        def usesKotlinPlugin = project.plugins.findPlugin('kotlin-android') != null
+        def hasAnnotationProcessorConfiguration = project.getConfigurations().findByName('annotationProcessor') != null
+        // TODO add a parameter in 'realm' block if this should be specified by users
+        def preferAptOnKotlinProject = false
```

Comment:
In that case, this implementation (and of course previous implementation too) expects users manually apply `android-apt` plugin.

If we need to provide a way for users to choose `annotationProcessor` configuration in Kotlin project, exposing config parameter is mandatory.